server/src/test/java/org/elasticsearch/index/mapper/ParametrizedMapperTests.java

```diff
@@ -296,116 +296,165 @@ private String toStringWithDefaults(ToXContent value) throws IOException {
+              "field": {
+                "type": "test_mapper",
+                "fixed": true,
+                "fixed2": false,
+                "variable": "default",
+                "index": true,
+                "wrapper": "default",
+                "int_value": 5,
+                "analyzer": "_keyword",
+                "required": "value",
+                "restricted": "foo",
+                "enum_field": "name1",
+                "restricted_enum_field": "name1"
+              }
+            }"""), toStringWithDefaults(mapper));
     }
 
     // merging - try updating 'fixed' and 'fixed2' should get an error, try updating 'variable' and verify update
     public void testMerging() {
-        String mapping = "{\"type\":\"test_mapper\",\"fixed\":false,\"required\":\"value\"}";
+        String mapping = """
+            {"type":"test_mapper","fixed":false,"required":"value"}""";
         TestMapper mapper = fromMapping(mapping);
         assertEquals("{\"field\":" + mapping + "}", Strings.toString(mapper));
 
-        TestMapper badMerge = fromMapping("{\"type\":\"test_mapper\",\"fixed\":true,\"fixed2\":true,\"required\":\"value\"}");
+        TestMapper badMerge = fromMapping("""
+            {"type":"test_mapper","fixed":true,"fixed2":true,"required":"value"}""");
         IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> mapper.merge(badMerge));
-        String expectedError = "Mapper for [field] conflicts with existing mapper:\n"
-            + "\tCannot update parameter [fixed] from [false] to [true]\n"
-            + "\tCannot update parameter [fixed2] from [false] to [true]";
+        String expectedError = """
+            Mapper for [field] conflicts with existing mapper:
+            \tCannot update parameter [fixed] from [false] to [true]
+            \tCannot update parameter [fixed2] from [false] to [true]""";
         assertEquals(expectedError, e.getMessage());
 
         assertEquals("{\"field\":" + mapping + "}", Strings.toString(mapper));   // original mapping is unaffected
 
         // TODO: should we have to include 'fixed' here? Or should updates take as 'defaults' the existing values?
-        TestMapper goodMerge = fromMapping("{\"type\":\"test_mapper\",\"fixed\":false,\"variable\":\"updated\",\"required\":\"value\"}");
+        TestMapper goodMerge = fromMapping("""
+            {"type":"test_mapper","fixed":false,"variable":"updated","required":"value"}""");
         TestMapper merged = (TestMapper) mapper.merge(goodMerge);
 
-        assertEquals("{\"field\":" + mapping + "}", Strings.toString(mapper));   // original mapping is unaffected
-        assertEquals(
-            "{\"field\":{\"type\":\"test_mapper\",\"fixed\":false,\"variable\":\"updated\",\"required\":\"value\"}}",
-            Strings.toString(merged)
-        );
+        assertEquals("""
+            {"field":%s}\
+            """.formatted(mapping), Strings.toString(mapper));   // original mapping is unaffected
```

Comment:
I think we should leave this line alone, the same line also appears above on line 344.
```suggestion
assertEquals("{\"field\":" + mapping + "}", Strings.toString(mapper)); // original mapping is unaffected
```